.github/actions/compile-examples/action.yml

```diff
@@ -1,17 +0,0 @@
-name: 'Compile Examples'
-author: 'Zachary J. Fields'
-description: 'Validate examples by compiling for supported platforms'
-runs:
-  using: 'docker'
-  image: './Dockerfile'
-  entrypoint: 'bash'
-  args:
-    - -c
-    - "cp -r . /home/blues/Arduino/libraries/Blues_Wireless_Notecard \
-      && HOME=/home/blues arduino-cli compile \
```

Comment:
This is the part you missed, and why you had to comment out the `USER` instruction from the Dockerfile (https://github.com/actions/runner/issues/1876).